docs/v3/release-notes/oss/version-3-4.mdx

```diff
@@ -2,6 +2,46 @@
 title: 3.4
 ---
 
+## 3.4.12 - 20k magic in the air
+
+*Released on August 08, 2025*
+
+**Bug Fixes 🐞**
+
+* Check future is if cancelled. by [@is](https://github.com/is) in [#18628](https://github.com/PrefectHQ/prefect/pull/18628)
+* Add in-memory causal ordering for task run recorder by [@desertaxle](https://github.com/desertaxle) in [#18634](https://github.com/PrefectHQ/prefect/pull/18634)
+* Add note on run_shell_script by [@rcash](https://github.com/rcash) in [#18658](https://github.com/PrefectHQ/prefect/pull/18658)
+* Note about `to_deployment` by [@zzstoatzz](https://github.com/zzstoatzz) in [#18647](https://github.com/PrefectHQ/prefect/pull/18647)
+* Fix event emission when blocks are loaded server-side by [@desertaxle](https://github.com/desertaxle) in [#18679](https://github.com/PrefectHQ/prefect/pull/18679)
+
+**Development & Tidiness 🧹**
+
+* chore(deps): bump ruff from 0.9.6 to 0.12.7 by [@dependabot](https://github.com/dependabot)[bot] in [#18622](https://github.com/PrefectHQ/prefect/pull/18622)
+* chore(deps): bump actions/download-artifact from 4 to 5 by [@dependabot](https://github.com/dependabot)[bot] in [#18650](https://github.com/PrefectHQ/prefect/pull/18650)
+* chore(deps): bump ruff from 0.12.7 to 0.12.8 by [@dependabot](https://github.com/dependabot)[bot] in [#18673](https://github.com/PrefectHQ/prefect/pull/18673)
+* chore(deps): bump uv from 0.5.7 to 0.8.6 in /scripts/proxy-test by [@dependabot](https://github.com/dependabot)[bot] in [#18672](https://github.com/PrefectHQ/prefect/pull/18672)
+
+**Documentation 📓**
+
+* Port examples hosting from PrefectHQ/examples by [@desertaxle](https://github.com/desertaxle) in [#18661](https://github.com/PrefectHQ/prefect/pull/18661)
+* Add Python SDK to variables-docs by [@oakhill87](https://github.com/oakhill87) in [#18626](https://github.com/PrefectHQ/prefect/pull/18626)
+* Make mention of Azure Event Grid support by [@bdalpe](https://github.com/bdalpe) in [#18631](https://github.com/PrefectHQ/prefect/pull/18631)
+* Fix link for ThreadPoolTaskRunner in Task Runners doc by [@bdalpe](https://github.com/bdalpe) in [#18663](https://github.com/PrefectHQ/prefect/pull/18663)
+* Bold the stray h2 by [@zzstoatzz](https://github.com/zzstoatzz) in [#18669](https://github.com/PrefectHQ/prefect/pull/18669)
```

Comment:
I think we should probably leave out super minor changes like this, but that's a loosely held opinion.